src/ray/protobuf/core_worker.proto

```diff
@@ -198,11 +198,22 @@ enum ObjectLocationState {
   ADDED = 0;
   // Object is removed.
   REMOVED = 1;
+  // Object is spilled.
+  SPILLED = 2;
```

Comment:
Instead of introducing a new state, is it possible to just have a nested message in `ObjectLocationStateUpdate` for spilling? Then we can keep using a single update for removed & spilled in `OwnershipBasedObjectDirectory::location_buffers_` and keep the logic to send only one update per object, but maybe it does not matter much.

Reply:
Yea, initially I was trying to make the code simpler. But I can change to a hashmap + a vector (for FIFO order), it shouldn't be too bad.